Python/thread_pthread.h

```diff
@@ -364,6 +365,15 @@ PyThread_exit_thread(void)
     pthread_exit(0);
 }
 
+void _Py_NO_RETURN
+_PyThread_pause_thread(void)
+{
+    dprintf(("PyThread_pause_thread called\n"));
+    while (1) {
+        pause();
```

Comment:
I'm assuming `pause` is subject to the signal mask as well, so I don't think it would change the behavior of other threads --- signals not blocked by the existing signal mask can already be delivered even without calling `pause`.  It seems like continuing to run signal handlers would be harmless.  What would the signal handler do that would be affected by Python finalizing?  It already cannot acquire the GIL or other locks.  I think blocking them would also probably be harmless, but that seems like potentially more interference with a thread that may not have been created by Python.

Reply:
> What would the signal handler do that would be affected by Python finalizing?

If an application embeds Python, it can continue to run and expect signals to continue working, after Python is finalized.

Reply:
Perhaps just use:

```
sigset_t set; /* we don't want to receive any signal */
sigfillset(&set);
sigsuspend(&set);
```

Instead of pause() or select() or poll() based sleeps.  It effectively combines the above pthread_sigmask with pause in one API call.